src/solvers/z3/z3_conv.cpp

```diff
@@ -1231,9 +1236,44 @@ void z3_smt_ast::dump() const
 
 void z3_convt::dump_smt()
 {
+  // #include <esbmc/bmc.h>
+  // #include <fstream>
+  const std::string &filename = options.get_option("output");
```

Comment:
Why are you creating a file?

Reply:
I needed a way to get the user-supplied filename for dumping the formulas directly to the file. A better approach could have been to overload the `dump_smt()` method that takes a `file` name string parameter (`std::string` type) but that would need changes all over the codebase and I didn't take that approach.